From 2aff2f4d4d4220ce15b00326c21a2828ed59c4b0 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 3 Sep 2024 17:03:08 +0200
Subject: [PATCH 1174/1695] drivers: st32_omm: consider pinctrl_apply_state()
 return code

Fix STM32 OM Manager driver to test pinctrl_apply_state() return code.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a379efc2d21d4634b3d92df97b4daadbd66555a9 drivers: st32_omm: consider pinctrl_apply_state() return code]
Change-Id: Ic5530f0bf639ff2b0bb02e46120896355138a3b1
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/401845
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_omm.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/core/drivers/stm32_omm.c b/core/drivers/stm32_omm.c
index 51a568f78..dfb23729e 100644
--- a/core/drivers/stm32_omm.c
+++ b/core/drivers/stm32_omm.c
@@ -328,8 +328,9 @@ static void stm32_omm_setup(void)
 {
 	stm32_omm_set_mm();
 	stm32_omm_configure();
-	if (omm_d->pinctrl_d)
-		pinctrl_apply_state(omm_d->pinctrl_d);
+	if (omm_d->pinctrl_d &&
+	    pinctrl_apply_state(omm_d->pinctrl_d))
+		panic();
 
 	if (firewall_set_alternate_conf(omm_d->firewall_conf))
 		panic();
@@ -337,8 +338,8 @@ static void stm32_omm_setup(void)
 
 static void stm32_omm_suspend(void)
 {
-	if (omm_d->pinctrl_s)
-		pinctrl_apply_state(omm_d->pinctrl_s);
+	if (omm_d->pinctrl_s && pinctrl_apply_state(omm_d->pinctrl_s))
+		panic();
 }
 
 static TEE_Result
-- 
2.47.2

